python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/router_bgp.py

```diff
@@ -53,12 +53,14 @@ def router_bgp(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol) -> None
 
             self.structured_config.router_bgp.neighbors.append_new(
                 ip_address=get_ip_from_ip_prefix(p2p_link_data["peer_ip"]),
-                remote_as=p2p_link_data["peer_bgp_as"],
+                remote_as=self.shared_utils.get_asn(p2p_link_data["peer_bgp_as"]),
                 peer=p2p_link_data["peer"],
                 description=p2p_link_data["peer"],
                 peer_group=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name if p2p_link.include_in_underlay_protocol else Undefined,
                 bfd=p2p_link.bfd,
-                local_as=p2p_link_data["bgp_as"] if p2p_link_data["bgp_as"] != self.shared_utils.bgp_as else None,
+                local_as=self.shared_utils.get_asn(p2p_link_data["bgp_as"])
+                if self.shared_utils.get_asn(p2p_link_data["bgp_as"]) != self.shared_utils.get_asn(self.shared_utils.bgp_as)
```

Comment:
I think we should have a cached property in shared_utils that is `final_bgp_as` (need a better name) rather to reconverting it again and again in multiple modules